src/ColorUtils.php

```diff
@@ -2,33 +2,134 @@
+    return $hex;
+  }
+  
+  
+  /**
+   * This PHP function calculates the brightness of a given colour in RGB format.
+   * 
+   * @param string colour The input colour in hexadecimal format (e.g. "#FF0000" for red).
+   * 
+   * @return float|int the brightness value of a given colour in the range of 0 to 255, where 0 is the
+   * darkest and 255 is the brightest. The brightness value is calculated using the formula for
+   * relative luminance, which takes into account the human eye's sensitivity to different colours.
+   */
+  public static function getBrightness(string $colour, bool $round = false): float|int {
+    $rgb = self::hexToRgb($colour);
+    $luminance = (($rgb[0] * 299) + ($rgb[1] * 587) + ($rgb[2] * 114)) / 1000;
+    return $round ? round($luminance) : $luminance;
+  }
+  
+  
+  /**
+   * The function determines if a given colour is dark or not based on its brightness value.
+   * 
+   * @param string colour The colour parameter is a string representing a colour in any valid CSS format
+   * (e.g. hex code, RGB, RGBA, HSL, HSLA).
+   * 
+   * @return bool A boolean value indicating whether the input colour is considered dark or not, based
+   * on its brightness level.
+   */
+  public static function isDark(string $colour): bool {
+    $brightness = self::getBrightness($colour);
+    return ($brightness < 128);
+  }
 
-        foreach ($rgba as $key => $value) {
-            $rgba[$key] = hexdec($value);
-        }
+  /**
+   * The function returns a contrasting color (either black or white) based on the darkness of the input
+   * color.
+   * 
+   * @param string colour The parameter "colour" is a string representing a color in hexadecimal format
+   * (e.g. "#FF0000" for red).
+   * 
+   * @return string a string value which is either "#FFFFFF" (white) or "#000000" (black) based on
+   * whether the input color is dark or not.
+   */
+  public static function getContrastColour(string $colour): string {
+    return self::isDark($colour) ? "#FFFFFF" : "#000000";
+  }
 
-        return $rgba;
-    }
-}
\ No newline at end of file
+
+  /**
+   * This function returns the complementary color of a given color in RGB or hexadecimal format.
+   * 
+   * @param colour The input color in hexadecimal format (e.g. "#FF0000" for red).
+   * @param hex The "hex" parameter is a boolean value that determines whether the function should
+   * return the complementary color in hexadecimal format (if set to true) or as an array of RGB values
+   * (if set to false).
+   * 
+   * @return the complementary color of the input color in either hexadecimal or RGB format, depending
+   * on the value of the  parameter.
+   */
+  public static function getComplementaryColour($colour, $hex = true): string|array {
+    $rgb = self::hexToRgb($colour);
+    $r = 255 - $rgb[0];
+    $g = 255 - $rgb[1];
+    $b = 255 - $rgb[2];
+    return $hex ? self::rgbToHex($r, $g, $b) : array($r, $g, $b);
+  } 
```

Comment:
```suggestion
  public static function getComplementaryColour($colour, $hex = true): string|array {
    $rgb = self::hexToRgb($colour);
    $r = 255 - $rgb[0];
    $g = 255 - $rgb[1];
    $b = 255 - $rgb[2];
    return $hex ? self::rgbToHex($r, $g, $b) : [$r, $g, $b];
  } 
```
For keeping things consistent please refrain from using `array(...)`

Reply:
You marked this as resolve but didn't change anything?